### PR TITLE
bluetooth: host: fix redundant reference count

### PR DESCRIPTION
**Pull request by xiaoliang314**
bt_hci_cmd_create creates a buffer with a reference count of 1, so there is no need to increase the reference count again. Test found that it caused the buffer not be released.

### CODE REVIEW

File: subsys/bluetooth/host/hci_core.c [hunk 1]
```diff
@@ -261,7 +261,7 @@ int bt_hci_cmd_send_sync(u16_t opcode, struct net_buf *buf,
 	cmd(buf)->sync = &sync_sem;
 
 	/* Make sure the buffer stays around until the command completes */
-	net_buf_ref(buf);
+	/* net_buf_ref(buf); */
```
Comment (carlescufi) [inline subsys/bluetooth/host/hci_core.c]: what about the case where the buffer is provided by the caller to `bt_hci_cmd_send_sync()`? you don't know what refcount is in that buf in that case.

Comment (xiaoliang314) [reply subsys/bluetooth/host/hci_core.c]: When bt_hci_cmd_send_sync() returns an error, the buffer will not be released, I think this is incorrect.

Comment (xiaoliang314) [reply subsys/bluetooth/host/hci_core.c]: Yeah, maybe you should use other methods to fix this problem.

**Review by cvinayak — CHANGES_REQUESTED** [review]

**Comment by xiaoliang314** [comment]
I think it can be fix like this:

	if (!buf) {
		buf = bt_hci_cmd_create(opcode, 0);
		if (!buf) {
			return -ENOBUFS;
		}
	} else {
		/* Make sure the buffer stays around until the command completes */
		net_buf_ref(buf);
	}

**Comment by carlescufi** [comment]
@xiaoliang314 can you update the PR with the code you posted in the review comment? Remember to amend the commit instead of adding a new commit.

**Comment by carlescufi** [comment]
@xiaoliang314 you are also missing a signed-off line in your commit message.
Use `git commit --amend -s` to add that.

**Comment by johnflux** [comment]
Xiao, why did you close this pull request?

**Comment by carlescufi** [comment]
@jhedberg do you  think this change in the code is needed?

**Pull request closed without merging**